internal/alloycli/cmd_run.go

```diff
@@ -398,19 +401,26 @@ func (fr *alloyRun) Run(cmd *cobra.Command, configPath string) error {
 	})
 
 	ready = f.Ready
-	reload = func() (*alloy_runtime.Source, error) {
-		alloySource, err := loadAlloySource(configPath, fr.configFormat, fr.configBypassConversionErrors, fr.configExtraArgs)
-		defer instrumentation.InstrumentConfig(err == nil, alloySource.SHA256(), fr.clusterName)
-
+	reload = func() (map[string][]byte, error) {
+		sources, err := loadSourceFiles(configPath, fr.configFormat, fr.configBypassConversionErrors, fr.configExtraArgs)
 		if err != nil {
+			instrumentation.InstrumentConfig(false, [32]byte{}, fr.clusterName)
```

Comment:
This seems like a regression, we're losing config metrics & the label on them. And we are no longer reporting it on success. Can we refactor the SHA256 function to work on raw sources, and make sure we're still calling this on non-failure case?

Reply:
Oh right missed success case :see_no_evil: 

>Can we refactor the SHA256 function to work on raw sources, and make sure we're still calling this on non-failure case?

Yeah I took a look at usage of `source.SHA256()` and this is the only place we use it so we could remove it from `Sources` and just do it here instead when we have gotten the raw sources from disk

Reply:
Fixed in [c1189ec](https://github.com/grafana/alloy/pull/3295/commits/c1189ecb29a45472a72f225b754ca3f90ff5a671)

I decided to not calculate the hash during parsing. The problem with having it there (and we had that problem before) is that if we ever got an error this would have been empty because we return `nil, error`.  The function was protected with (nil check)[https://github.com/grafana/alloy/blob/main/internal/runtime/source.go#L173-L175]